### PR TITLE
feat(X): Add `Open links as query` patch

### PR DESCRIPTION
**Pull request by dic1911**
As title, let me know if I missed anything.

### CODE REVIEW

File: src/main/kotlin/app/revanced/patches/twitter/misc/browserchooser/BrowserChooserPatch.kt [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            )
+        })
+
+        val inject = """
+            invoke-static {p0, p1}, $className->$helperMethodName(Landroid/content/Context;Landroid/content/Intent;)$returnType
+            return-void
+        """.trimIndent()
+
+        val instructions = method.getInstructions()
+        result.mutableMethod.removeInstructions(0, instructions.count())
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/twitter/misc/browserchooser/BrowserChooserPatch.kt:72]: Simply injecting at index 0 should suffice

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]
Looks good all in all. I'll push some small changes, until then you can look into the current review

File: src/main/kotlin/app/revanced/patches/twitter/misc/browserchooser/BrowserChooserPatch.kt [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            className,
+            helperMethodName,
+            listOf(ImmutableMethodParameter("Landroid/content/Context;", null, "context"), ImmutableMethodParameter("Landroid/content/Intent;", null, "intent")),
+            returnType,
+            AccessFlags.STATIC.value,
+            null,
+            null,
+            MutableMethodImplementation(5)
+        ).toMutable().apply {
+            addInstructions(
```
[the remark this answers is not in the document]
Comment (dic1911) [reply src/main/kotlin/app/revanced/patches/twitter/misc/browserchooser/BrowserChooserPatch.kt:48]: Agreed, here's the PR for that.
https://github.com/ReVanced/revanced-integrations/pull/570

File: src/main/kotlin/app/revanced/patches/twitter/misc/browserchooser/BrowserChooserPatch.kt (hunk 1 above)
Comment (dic1911) [reply src/main/kotlin/app/revanced/patches/twitter/misc/browserchooser/BrowserChooserPatch.kt]: Just force pushed with things cleaned up, tested on my side and seems to be working good.

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]
Pushed some changes, please test them. (You have to include the patch as `use = false`).
In the integrations I have also removed instantiating a new intent and instead overwrote the action. Is that even necessary?

**Comment by dic1911** [comment]
@oSumAtrIX the `--include`/`-i` flag doesn't work on my side, I can't test it with `use` set to false.

File: src/main/kotlin/app/revanced/patches/twitter/misc/links/OpenLinksWithAppChooserPatch.kt [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    use = false,
+)
+@Suppress("unused")
+object OpenLinksWithAppChooserPatch : BytecodePatch(
+    setOf(OpenLinkFingerprint),
+) {
+    private const val METHOD_REFERENCE =
+        "Lapp/revanced/integrations/twitter/patches/links/OpenLinksWithAppChooserPatch;->" +
+            "openWithChooser(Landroid/content/Context;Landroid/content/Intent;)V"
+    override fun execute(context: BytecodeContext) {
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/twitter/misc/links/OpenLinksWithAppChooserPatch.kt:25]:
```suggestion

    override fun execute(context: BytecodeContext) {
```

**Pull request merged**